partner_invoicing_mode_at_shipping/models/sale_order.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    one_invoice_per_shipping = fields.Boolean(
+        compute="_compute_one_invoice_per_shipping",
+        store=True,
+        index=True,
+    )
+
+    @api.depends("partner_invoice_id")
+    def _compute_one_invoice_per_shipping(self):
+        """
+        Compute this field (instead a related) to avoid computing all
+        related sale orders if option changed on partner level.
+        """
+        for order in self:
+            order.one_invoice_per_shipping = (
+                order.partner_invoice_id.one_invoice_per_shipping
+            )
+
+    def _get_generated_invoices(self, partition):
+        sales_to_validate_invoices = self.filtered(
+            lambda sale: sale.one_invoice_per_shipping
+            and any(invoice.state == "draft" for invoice in self.invoice_ids)
+        )
```

Comment:
@rousseldenis But if the invoice is already created and in draft, then the sale is not anymore in status "to invoice" and will be excluded from self with `_get_generate_invoices_state_domain`.

How can you enter this method? I think you won't even have a job for such sales

Reply:
The validation of those is already tackled by your `_validate_per_shipping_generated_invoices`, then I don't get why you need this hook ?

Reply:
I check this. It was part of the former implementation.